subprojects/core/src/integTest/groovy/org/gradle/api/tasks/LambdaInputsIntegrationTest.groovy

```diff
@@ -373,16 +492,35 @@ class LambdaInputsIntegrationTest extends AbstractIntegrationSpec implements Val
             import java.io.File;
             import java.io.IOException;
             import java.nio.file.Files;
+            import java.util.function.Consumer;
 
             public class ${className} {
 ${classBody}
             }
         """
     }
 
-    private static String lambdaWritingFile(String constantName, String outputString) {
+    /**
+     * Configuration cache infrastructure makes sure that {@code Action} instances are always serializable.
+     */
+    private static String serializableActionLambdaWritingFile(String constantName, String outputString) {
```

Comment:
```suggestion
    private static String actionLambdaWritingFile(String constantName, String outputString) {
```

Maybe we should drop the `serializable` from the method name, given that `Action` is not serializable from the start. With the comment I think it would make sense.

Reply:
Done